defold-rive/commonsrc/atlas.cpp

```diff
@@ -204,35 +205,76 @@ namespace dmRive {
         }
     }
 
+    rive::rcp<rive::RenderImage> LoadImageFromFactory(dmResource::HFactory factory, HRenderContext context, const char* path)
+    {
+        rive::rcp<rive::RenderImage> image;
+        if (!factory)
+        {
+            dmLogError("No factory provided!");
+            return image;
+        }
+
+        char path_buffer[256];
+        uint32_t path_length = 0;
+        if (path[0] != '/')
+            path_buffer[path_length++] = '/';
+        path_length += dmStrlCpy(&path_buffer[path_length], path, sizeof(path_buffer)-path_length);
+
+        void* resource;
+        uint32_t resource_size;
+        //dmResource::Result r = dmResource::Get(factory, path_buffer, &resource);
+        dmResource::Result r = dmResource::GetRaw(factory, path_buffer, &resource, &resource_size);
+        if (dmResource::RESULT_OK != r)
+        {
+            dmLogError("Error getting file '%s': %d", path_buffer, r);
+            return image;
+        }
+
+        dmhash_t name_hash = dmHashString64(path_buffer);
+
+        image = CreateRiveRenderImage(context, resource, resource_size);
+
+        return image;
+    }
 
-    AtlasNameResolver::AtlasNameResolver(HRenderContext context)
-    : m_RiveRenderContext(context)
+    AtlasNameResolver::AtlasNameResolver(dmResource::HFactory factory, HRenderContext context)
+    : m_Factory(factory)
+    , m_RiveRenderContext(context)
     {
     }
 
     bool AtlasNameResolver::loadContents(rive::FileAsset& _asset, rive::Span<const uint8_t> inBandBytes, rive::Factory* factory)
     {
+        DEBUGLOG("loadContents");
+
+        if (m_Assets.Full())
+        {
+            m_Assets.OffsetCapacity(8);
+        }
+        m_Assets.Push(&_asset);
+
+        bool out_of_band = inBandBytes.size() == 0;
+            if (inBandBytes.size() == 0) // out-of-band asset;
```

Comment:
Indentation?

Reply:
Yeah, some temp copy-paste?!

Not only looks weird, but also changes the logic!!! 😱